bats/tests/profile/create-profile-output.bats

```diff
@@ -0,0 +1,424 @@
+}
+
+@test 'complains when no output type is specified' {
+    run rdctl create-profile --from-settings
+    assert_failure
+    assert_output --partial 'an "--output FORMAT" option of either "plist" or "reg" must be specified'
+}
+
+@test 'complains when an invalid output type is specified' {
+    run rdctl create-profile --from-settings --output=cabbage
+    assert_failure
+    assert_output --partial 'received unrecognized "--output FORMAT" option of "cabbage"; "plist" or "reg" must be specified'
+}
+
+@test 'complains when no input source is specified' {
+    for type in reg plist; do
+        run rdctl create-profile --output $type
+        assert_failure
+        assert_output --partial 'no input format specified: must specify exactly one input format of "--input FILE|-", "--body|-b STRING", or "--from-settings"'
+    done
+}
+
+@test 'complains when no --input arg is specified' {
+    for type in reg plist; do
+        for input in input body; do
+            run rdctl create-profile --output "$type" --"$input"
+            assert_failure
+            assert_output --partial $"Error: flag needs an argument: --$input"
+        done
+    done
+}
+
+too_many_input_formats() {
+    run rdctl create-profile "$@"
+    assert_failure
+    assert_output --partial 'too many input formats specified: must specify exactly one input format of "--input FILE|-", "--body|-b STRING", or "--from-settings"'
+}
+
+@test 'complains when multiple input sources are specified' {
+    for type in reg plist; do
+        too_many_input_formats --output $type --input some-file.txt -b moose
+        too_many_input_formats --output $type --input some-file.txt --from-settings
+        too_many_input_formats --output $type --input some-file.txt -b moose --from-settings
+        too_many_input_formats --output $type -b moose --from-settings
+    done
+}
+
+@test "complains when input file doesn't exist" {
+    run rdctl create-profile --output reg --input /no/such/file/here
+    assert_failure
+    assert_output --partial 'Error: open /no/such/file/here:'
+}
+
+@test 'report invalid parameters for plist' {
+    run rdctl create-profile --output=plist --from-settings --hive=fish
+    assert_failure
+    assert_output --partial $"registry hive and type can't be specified with \"plist\""
+
+    run rdctl create-profile --output plist --from-settings --type=writer
+    assert_failure
+    assert_output --partial $"registry hive and type can't be specified with \"plist\""
+}
+
+@test 'report unrecognized output-options' {
+    run rdctl create-profile --output=pickle
+    assert_failure
+    assert_output --partial 'received unrecognized "--output FORMAT" option of "pickle"; "plist" or "reg" must be specified'
+}
+
+@test 'report unrecognized registry sub-options' {
+    run rdctl create-profile --output=reg --hive=hklm --type=ruff --from-settings
+    assert_failure
+    assert_output --partial 'invalid registry type of "ruff" specified'
+}
```

Comment:
Should we have a corresponding test for invalid `--hive` argument?

Reply:
sure